packages/klighd-core/src/scaling-util.ts

```diff
@@ -0,0 +1,203 @@
+        // result_b = scaleDimension(offset_b, length_b, available, scale)
+        // result_a.offset = result_b.offset + result_b.length + margin
+        // || result_b.offset = result_a.offset + result_a.length + margin
+
+        const fa = (offset_a * length_a) / (available - length_a)
+        const fb = (offset_b * length_b) / (available - length_b)
+
+        const numerator = offset_a + fa - offset_b - fb
+
+        const result_1 = ( numerator - margin) / (fa - fb + length_b)
+        const result_2 = (-numerator - margin) / (fb - fa + length_a)
+
+        // the scale should be at least one and at most one of the results should be positive
+        return Math.max(result_1, result_2, 1)
+    }
+
+
+    /**
+     * Calculate the maximum scale at which node and sibling retain the given margin between them
+     * @param node the bounds of node to scale
+     * @param parent the dimensions of the parent of the node to scale
+     * @param sibling a sibling og the nod to scale
+     * @param margin the margin node and sibling shall retain when both scaled by the result
+     * @returns the maximum scale at which node and sibling retain margin between them
+     */
+    public static maxSiblingScale(node: Bounds, parent: Dimension, sibling: Bounds, margin: number) : number {
+
+        // calculate the scale for each dimension at which we reach our sibling
+        const result_1 = ScalingUtil.inverseScaleDimension(node.x, node.width, sibling.x, sibling.width, parent.width, margin)
+        const result_2 = ScalingUtil.inverseScaleDimension(node.y, node.height, sibling.y, sibling.height, parent.height, margin)
+
+        // take the max as that which ever is further is relevant for bounding us, but should be at least 1
+        return Math.max(result_1, result_2, 1)
+    }
+    /**
+     * Scale bounds in the specified dimensions by the specified scale
+     * @param originalBounds the bounds to scale
+     * @param availableSpace the space available to scale the bounds
+     * @param scale the scale by which to scale
+     * @returns the scaled bounds
+     */
+    private static calculateScaledBounds(originalBounds: Bounds, availableSpace: Dimension, scale: number) : Bounds {
+        const originalWidth = originalBounds.width
+        const originalHeight = originalBounds.height
+        const originalX = originalBounds.x
+        const originalY = originalBounds.y
+
+        // Calculate the new x offset and width:
+        const {length: newWidth, offset: newX} = ScalingUtil.scaleDimension(originalX, originalWidth, availableSpace.width, scale)
+
+        // Same for y offset and height
+        const {length: newHeight, offset: newY} = ScalingUtil.scaleDimension(originalY, originalHeight, availableSpace.height, scale)
+
+        return {x: newX, y : newY, width: newWidth, height: newHeight}
+    }
+
+    private static scaleDimension(offset: number, length: number, available: number, scale: number) : {offset:number, length:number}{
```

Comment:
My version of this was commented so nicely to explain what the code does, why did you not keep that?

Reply:
re-added the dropped comments in 7b68e03b, slightly different wording as this function is abstract over with and height